packages/shared/src/components/modals/SubmitArticle.spec.tsx

```diff
@@ -0,0 +1,164 @@
+
+const onRequestClose = jest.fn();
+
+beforeEach(async () => {
+  nock.cleanAll();
+  jest.clearAllMocks();
+});
+
+const defaultUser: LoggedUser = {
+  id: 'u1',
+  name: 'Ido Shamun',
+  providers: ['github'],
+  email: 'ido@acme.com',
+  image: 'https://daily.dev/ido.png',
+  infoConfirmed: true,
+  premium: false,
+  createdAt: '2020-07-26T13:04:35.000Z',
+  username: 'test',
+  permalink: 'sample',
+};
+
+const renderComponent = (
+  isEnabled = false,
+  user: LoggedUser | AnonymousUser = defaultUser,
+): RenderResult => {
+  const client = new QueryClient();
+  return render(
+    <QueryClientProvider client={client}>
+      <AuthContextProvider
+        user={user}
+        updateUser={jest.fn()}
+        tokenRefreshed
+        getRedirectUri={jest.fn()}
+        loadingUser={false}
+        loadedUserFromCache
+      >
+        <SubmitArticle
+          headerCopy="Submit article"
+          submitArticleModalButton="Submit article"
+          isEnabled={isEnabled}
+          isOpen
+          onRequestClose={onRequestClose}
+        />
+      </AuthContextProvider>
+    </QueryClientProvider>,
+  );
+};
+
+it('should show request access for certain users', async () => {
+  renderComponent();
+  expect(await screen.findByLabelText('Request access')).toBeInTheDocument();
+});
+
+it('should show a message no URL was set', async () => {
+  renderComponent(true);
+  const btn = await screen.findByLabelText('Submit article');
+  btn.click();
+  expect(
+    await screen.findByText('Please submit a valid URL'),
+  ).toBeInTheDocument();
+});
+
+it('should disable the button on invalid URL', async () => {
+  renderComponent(true);
+  const input = await screen.findByRole('textbox');
+  userEvent.type(input, 'fakeURL');
+  const btn = await screen.findByLabelText('Submit article');
+  expect(btn).toBeDisabled();
+});
+
+it('should submit a valid URL', async () => {
+  renderComponent(true);
+  const input = await screen.findByRole('textbox');
+  userEvent.type(input, 'http://blog.daily.dev/blog/article-1');
+  const btn = await screen.findByLabelText('Submit article');
+  btn.click();
+
+  mockGraphQL({
+    request: {
+      query: SUBMIT_ARTICLE_MUTATION,
+      variables: { url: 'http://blog.daily.dev/blog/article-1' },
+    },
+    result: {
+      data: {
+        submitArticle: {
+          _: true,
+        },
+      },
+    },
+  });
+
+  expect(await screen.findByText('Request sent')).toBeInTheDocument();
+});
+
+it('should feedback existing article', async () => {
+  renderComponent(true);
+  const input = await screen.findByRole('textbox');
+  userEvent.type(input, 'http://blog.daily.dev/blog/article-1');
+  const btn = await screen.findByLabelText('Submit article');
+  btn.click();
+
+  mockGraphQL({
+    request: {
+      query: SUBMIT_ARTICLE_MUTATION,
+      variables: { url: 'http://blog.daily.dev/blog/article-1' },
+    },
+    result: {
+      data: {
+        submitArticle: null,
+      },
+      errors: [
```

Comment:
I suggested changing this behavior in the API PR. GraphQL errors are not type-safe and thus I'm not a fan of using them heavily.

Reply:
This is changed to the API